### DISCUSSION TITLE
backend.ai-client-py의 image rescan 관련 질문입니다.

### DISCUSSION THREAD

**Discussion by TeusEE** [lablup/contribution-academy-2021#95] [Q&A]
관련 Issue : https://github.com/lablup/backend.ai/issues/67

해당 Issue 해결을 위해서
backend.ai-client-py/src/ai/backend/client/cli/admin/image.py(https://github.com/lablup/backend.ai-client-py/blob/main/src/ai/backend/client/cli/admin/image.py#L47)
를 확인 중 입니다.

현재 아래 순서로 코드 확인중에 막혀있습니다.
backend.ai-client-py/src/ai/backend/client/cli/admin/image.py(CLI Input)
-> backend.ai-client-py/src/ai/backend/client/func/image.py
-> backend.ai-client-py/src/ai/backend/client/session.py
-> backend.ai-client-py/src/ai/backend/client/func/admin.py
-> backend.ai-client-py/src/ai/backend/client/requests.py
-> ????

reqeusts.py에서 FetchContextManager obejct를 return하게 되어있는데,
이때 GraphQL Request가 어느 서버로 전달되고, Response를 받아오는지 확인하는 부분에서 어려움을 겪고 있습니다.

https://github.com/lablup/backend.ai-client-py/blob/b6a2bd91c7b57c29ce86d0a6bdce529a16a0260e/src/ai/backend/client/request.py#L561
해당 위치에서 async로 response를 받는 것으로 예상되는데,  Cli의 추가를 위해서 Reqeust를 처리해주는 Server File의 위치를 알 수 있으면 좋을 것 같습니다.

**Comment by youngjun0627** [comment]
1. client는 rescan-image명령을 manager에게 request한다.
2. manager는 client로부터 받은 request를 토대로 task를 생성한다.
3. 해당 task의 id를 클라이언트에게 response한다.
4. 클라이언트는 해당 response를 이용하여 태스크를 추적한다.

보니깐 rescan-images명령이 백그라운드로 실행되는 것으로 확인되었습니다. [manager-rescan](https://github.com/lablup/backend.ai-manager/blob/85fe578508ad2edb6cd44587b607157a8d43ae76/src/ai/backend/manager/models/image.py#L221)에서 task를 생성하는 것을 확인하실 수 있습니다.

**Reply by TeusEE** [reply]
감사합니다! background task로 돌리는거는 생각도 못했네요!

**Comment by lizable** [comment]
@TeusEE Could you tell me about the changes you made in `container_registry/base.py` from manager repo in detail?
For now, I need additional contexts that you tried to change. I can't see any PR related, so It's hard to track down what you've been struggling with.

**Reply by lizable** [reply]
And could you share the strategy about distinguishing "update" / "delete" / "keep-current" images?
